File: 1404. Number of Steps to Reduce a Number in Binary Representation to One/numSteps.js
/**
 * @param {string} s
 * @return {number}
 */
const numSteps = (s) => {
  let steps = 0;
  let carry = 0;

  for (let i = s.length - 1; i > 0; i--) {
    if (parseInt(s[i]) + carry === 1) {
      carry = 1;
      steps += 2;
    } else {
      steps += 1;
    }
  }

  return steps + carry;
};


const s = "1101";
console.log('numSteps: ', numSteps(s))

File: 1404. Number of Steps to Reduce a Number in Binary Representation to One/numSteps.py
class Solution:
  def numSteps(self, s: str) -> int:
    steps = 0
    carry = 0
    
    for i in range(len(s) - 1, 0, -1):
      if int(s[i]) + carry == 1:
        carry = 1
        steps += 2
      else:
        steps += 1
        
    return steps + carry
  
  
s = "1101"
solution = Solution()
print('numSteps: ', solution.numSteps(s))
